### PR TITLE
Post xmas fixes

### PR DESCRIPTION
**Pull request by mtewes**
Changes discussed in the issues... for me it's OK to merge this in.

### CODE REVIEW

**Comment by mtewes** [comment]
@kuntzer , instead of viewing the code, if you want we can also briefly skype!

File: tenbilac/com.py [hunk 1]
```diff
@@ -149,14 +164,14 @@ def train(self, inputs, targets, inputnames=None, targetnames=None):
...
-					name='{}-{}-{}'.format(nettype, self.name, i)
+					name='{}_{}'.format(self.name, i)
 					)
 			else:
 				raise RuntimeError("Don't know network type '{}'".format(nettype))
 			
 			
 			# A directory where the training can store its stuff
-			trainobjdir = os.path.join(self.workdir, "{}_{:03d}".format(self.name, i))
+			trainobjdir = os.path.join(self.workdir, "{}_{}".format("member", i))
```
Comment (kuntzer) [inline tenbilac/com.py]: `"{}_{}".format("member", i)` it's a bit of an overkill, no? better would be: `"member_{}".format(i)`

File: demo/com/tenbilac.cfg [hunk 2]
```diff
@@ -10,14 +10,26 @@
...
-workdir: ./workdir
-#workdir: /vol/fohlen11/fohlen11_1/mtewes/tenbilac_demo_workdir
+# Note that MegaLUT supercedes this setting.
+#workdir: ./workdir
+workdir: /vol/fohlen11/fohlen11_1/mtewes/tenbilac_demo_workdir
 
+# Should the config be written into the workdir when starting a training (for future reference)?
 copyconfig: True
 
+# Should a minimized version of the committee be created after each training (useful to send to a friend)?
```
Comment (kuntzer) [inline demo/com/tenbilac.cfg]: Just to improve the doc string: what does "minimized" mean?

File: tenbilac/com.py [hunk 3]
```diff
@@ -400,6 +462,15 @@ def predict(self, inputs):
 		
 
 
+def datetimestr(dt):
```
Comment (kuntzer) [inline tenbilac/com.py]: Maybe, this could be moved to a more general tool file, like dirs or io ?

File: tenbilac/net.py [hunk 4]
```diff
@@ -109,10 +109,20 @@ def report(self):
 		return "\n".join(txt)
 
 	
-	def save(self, filepath):
+	def resetcache(self):
```
Comment (kuntzer) [inline tenbilac/net.py]: Nice :)

File: tenbilac/train.py [hunk 5]
```diff
@@ -26,21 +26,24 @@ class Training:
 	"""
 
 	
-	def __init__(self, net, dat, errfctname="msrb", regulweight=None, regulfctname=None, itersavepath=None, saveeachit=False, autoplotdirpath=".", autoplot=False, trackbiases=False, verbose=False, name=None):
+	def __init__(self, net, dat, errfctname="msb", regulweight=None, regulfctname=None, itersavepath=None, saveeachit=False, autoplotdirpath=".", autoplot=False, trackbiases=False, verbose=False, name=None):
```
Comment (kuntzer) [inline tenbilac/train.py]: Maybe we should force the user to select an `errfctname`, no? (i.e. no default)

File: tenbilac/train.py [hunk 6]
```diff
@@ -49,13 +52,14 @@ def __init__(self, net, dat, errfctname="msrb", regulweight=None, regulfctname=N
...
-		else:
-			logger.warning("Unknown error function, will blindly go ahead...")
+		if dat != None:
+			assert net.ni == self.dat.getni()
+			if errfctname in ["mse", "msb", "msrb", "msre", "msbw"]:
+				assert net.no == self.dat.getno()
+			elif errfctname in ["msbwnet"]:
+				assert net.no == 2*self.dat.getno()
+			else:
+				logger.warning("Unknown error function, will blindly go ahead...")
```
Comment (kuntzer) [inline tenbilac/train.py]: This will very likely fail anyway, yes?

File: tenbilac/com.py (hunk 1 above)
Comment (mtewes) [inline tenbilac/com.py]: Indeed, will change

File: tenbilac/com.py (hunk 3 above)
Comment (mtewes) [inline tenbilac/com.py]: Yes, but com.py is the only "interface" of Tenbilac that will ever care about such things, so I leave this one here.

File: demo/com/tenbilac.cfg (hunk 2 above)
Comment (mtewes) [inline demo/com/tenbilac.cfg]: will fix

File: tenbilac/train.py (hunk 5 above)
Comment (mtewes) [inline tenbilac/train.py]: We could. But to use Training merely as a container (as we do here), it's nice to have defaults.

File: tenbilac/train.py (hunk 6 above)
Comment (mtewes) [inline tenbilac/train.py]: Yep.

**Pull request merged**